packages/mobile/src/services/offline-downloader/offline-sync.ts

```diff
@@ -0,0 +1,253 @@
+      trackId: addedTrack,
+      downloadReason: {
+        is_from_favorites: true,
+        collection_id: DOWNLOAD_REASON_FAVORITES
+      }
+    })
+  )
+  batchDownloadTrack(tracksForDownload)
+
+  const tracksForDelete: TrackForDownload[] = removedTrackIds.map(
+    (removedTrack) => ({
+      trackId: removedTrack,
+      downloadReason: {
+        is_from_favorites: true,
+        collection_id: DOWNLOAD_REASON_FAVORITES
+      }
+    })
+  )
+  batchRemoveTrackDownload(tracksForDelete)
+}
+
+const syncCollection = async (
+  offlineCollection: Collection,
+  isFavoritesDownload?: boolean
+) => {
+  // TODO: record and check last verified time for collections
+  const state = store.getState()
+  const currentUserId = getUserId(state as unknown as CommonState)
+  const collectionId = offlineCollection.playlist_id
+  const collectionIdStr = offlineCollection.playlist_id.toString()
+  const updatedCollection: UserCollectionMetadata | undefined = (
+    await apiClient.getPlaylist({
+      playlistId: collectionId,
+      currentUserId
+    })
+  )?.[0]
+
+  // TODO: will discovery serve a removed playlist?
+  if (!updatedCollection) return
+
+  if (
+    moment(updatedCollection.updated_at).isSameOrBefore(
+      offlineCollection.updated_at
+    )
+  ) {
+    return
+  }
+
+  const updatedCollectionWithArt = await populateCoverArtSizes(
+    updatedCollection
+  )
+
+  downloadCollection(updatedCollectionWithArt)
+  if (
+    updatedCollectionWithArt.cover_art_sizes !==
+    offlineCollection.cover_art_sizes
+  ) {
+    downloadCollectionCoverArt(updatedCollectionWithArt)
+  }
+
+  const oldTrackIds = new Set(
+    offlineCollection.tracks?.map((track) => track.track_id)
+  )
+  const newTrackIds = new Set(
+    updatedCollection.tracks?.map((track) => track.track_id)
+  )
+  const addedTrackIds = [...newTrackIds].filter(
+    (trackId) => !oldTrackIds.has(trackId)
+  )
+  const removedTrackIds = [...oldTrackIds].filter(
+    (trackId) => !newTrackIds.has(trackId)
+  )
+
+  const tracksForDelete: TrackForDownload[] = removedTrackIds.map(
+    (removedTrack) => ({
+      trackId: removedTrack,
+      downloadReason: {
+        is_from_favorites: isFavoritesDownload,
+        collection_id: collectionIdStr
+      }
+    })
+  )
+  removeCollectionDownload(collectionIdStr, tracksForDelete)
+
+  const tracksForDownload: TrackForDownload[] = addedTrackIds.map(
+    (addedTrack) => ({
+      trackId: addedTrack,
+      downloadReason: {
+        is_from_favorites: isFavoritesDownload,
+        collection_id: collectionIdStr
+      }
+    })
+  )
+  batchDownloadTrack(tracksForDownload)
+}
+
+const syncStaleTracks = () => {
+  const state = store.getState()
+  const cacheTracks = getTracks(state, {})
+  const currentUserId = getUserId(state as unknown as CommonState)
+  if (!currentUserId) return
+
+  const staleCachedTracks = Object.entries(cacheTracks)
+    .filter(
+      ([id, track]) =>
+        track.offline &&
+        moment()
```

Comment:
cool! not that we are making this push at all rn, but im hoping we get off moment at some point. do you have any alternative date libraries you've used in past? ive personally used dayjs cause it's so small and pretty effective

Reply:
oh, nice. No, I'm just familiar with moment and js Date